app/src/main/java/com/example/android_level_3/FragmentAddContact.kt

```diff
@@ -0,0 +1,219 @@
+
+    private fun requestGetAllUsers() {
+        viewModel.getAllUsers(token = viewModel.token.value!!, progressBar = isVisibleProgressBar)
+    }
+
+    private fun setListeners() {
+
+        binding.toolbarAddContact.imgBackAddContact.setOnClickListener {
+            findNavController().popBackStack()
+        }
+
+        binding.toolbarAddContact.imgSearchAddContact.setOnClickListener {
+            viewModel.isActiveSearchAddContact.value = true
+        }
+
+        binding.toolbarAddContact.imgCloseAddContact.setOnClickListener {
+            viewModel.isActiveSearchAddContact.value = false
+        }
+
+        binding.toolbarAddContact.edSearchAddContact.addTextChangedListener(object : TextWatcher {
+            override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) { }
+            override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) { }
+            override fun afterTextChanged(s: Editable?) {
+                if(s?.isNotEmpty() == true ) {
+                    viewModel.filteredUserList.value?.clear()
+
+                    val filteredList = viewModel.listOfAllUsers.value?.filter {
+                        it.name?.contains(s, true) == true
+                    }
+                    viewModel.filteredUserList.value = filteredList?.toMutableList()
+                    afterFilterAction()
+                } else {
+                    if (viewModel.listOfAllUsers.value?.isNotEmpty() == true)
+                        viewModel.listOfAllUsers.value = viewModel.listOfAllUsers.value
+                }
+            }
+        })
+    }
+
+    private fun afterFilterAction() {
+        if (viewModel.filteredUserList.value?.isNotEmpty() == true) {
+            binding.recyclerViewContacts.visibility = View.VISIBLE
+            binding.noContactsContainer.visibility = View.GONE
+            if (this::recyclerViewAdapter.isInitialized) {
+                recyclerViewAdapter.submitList(viewModel.filteredUserList.value?.toList())
+            } else {
+                createAdapter( multiSelectState = null,
+                    usersInContactList = viewModel.userContactsIdList.toList(),
+                    filteredContactsList = viewModel.filteredUserList.value?.toList())
+            }
+        } else {
+            binding.recyclerViewContacts.visibility = View.GONE
+            binding.noContactsContainer.visibility = View.VISIBLE
+        }
+
+    }
+    private fun createConnectionErrorSnackBar(requestType: String): Snackbar {
+        return Snackbar.make(binding.root, "Problem with connection...", Snackbar.LENGTH_INDEFINITE)
+            .setActionTextColor(requireActivity().getColor(R.color.orange_color))
+            .setAction("RETRY") {
+                when(requestType) {
+                    Const.REQUEST_GET_ALL_USERS -> { requestGetAllUsers() }
+                    Const.REQUEST_ADD_CONTACT -> { requestAddToContactList() }
+                }
+            }
+    }
```

Comment:
All the info user would see should be in the resources. 
Also, I don't understand the necessity to return the Snackbar like this. I would rather have a function which displays some text than returning the Snackbar instance.